src/pages/api/event/[id]/repairers.ts

```diff
@@ -0,0 +1,44 @@
+import { NextApiRequest, NextApiResponse } from "next";
+import { ApiError } from "next/dist/server/api-utils";
+import { getAuth } from "@clerk/nextjs/server";
+import { HttpStatusCode } from "axios";
+
+import apiHandler from "@/lib/api-handler";
+import { EventRepairer } from "@/types";
+
+import prisma from "../../../../lib/prisma";
+
+export default apiHandler({
+  post: createRepairer
+});
+
+async function createRepairer(
+  req: NextApiRequest,
+  res: NextApiResponse<EventRepairer>
+) {
+  const { userId } = req.body;
+  const { id } = req.query;
+
+  const event = await prisma.event.findUnique({
+    where: { id: id as string }
+  });
+
+  if (!event) {
+    throw new ApiError(HttpStatusCode.NotFound, "Event not found");
+  }
+
+  const user = await getAuth(req);
```

Comment:
What if the user here doesn't match the userId provided in the request body? An attacker can spoof the userId so we should use this variable instead

Reply:
@justinliangg @dct0 Could somebody tell me what is the actual usage of this api? If we are just going to add the current user to the event, we wouldn't have to send userid in body because we can get it from session. If this is a api to add any user to event, we might have to validate current user have proper permission (in specific role).

Reply:
This API is to add a repairer to an event. 

Yes, @Wolfgang1208 will need to check the userId in the request body if it exists in Clerk. Have a look at the `getUser` function in [user service ](https://github.com/codersforcauses/repair-labs/blob/main/src/services/user.ts) to perform the check. The `getAuth` function is not used correctly as that gets the session of the user that queried the endpoint 

As for the validating the current user has the proper permission to perform the action, that will be in a separate issue to implement RBAC to add authorisation to the APIs.